rules-reviewed/jboss-eap/websphere/websphere-mq.windup.xml

```diff
@@ -0,0 +1,846 @@
+                </classification>
+                <hint>
+                    <message>In JBoss EAP 6, the JMS destinations can be controlled with JBoss CLI.</message>
+                    <link description="JBoss CLI" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.4/html-single/Administration_and_Configuration_Guide/index.html#Configure_the_JMS_Server1"/>
+                </hint>
+            </perform>
+        </rule>
+
+
+
+        <!-- WebSphere JMS MQ -->
+
+        <!-- Package-matching generic rule - MQ. -->
+        <rule id="websphere-mq-02500">
+            <when>
+                <javaclass references="com.ibm.mq.jms.{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ API">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/package-summary.html"/>
+                </classification>
+            </perform>
+        </rule>
+
+        <!-- Package-matching generic rule - MQe. -->
+        <rule id="websphere-mq-02501">
+            <when>
+                <javaclass references="com.ibm.mqe.jms.{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQe API">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/package-summary.html"/>
+                </classification>
+            </perform>
+        </rule>
+
+        <!-- Example code to match:
+            Properties env = new Properties();
+            env.put(
+                Context.INITIAL_CONTEXT_FACTORY,
+                "com.ibm.mq.jms.context.WMQInitialContextFactory");
+        -->
+        <rule id="websphere-mq-03000">
+            <when>
+                <!-- WINDUP-615
+                <or>
+                    <javaclass references="com.ibm.mq.jms.context.WMQInitialContextFactory" >
+                        <location>STRING</location>
+                    </javaclass>
+                    <filecontent filename="{*}.properties" pattern="{*}com.ibm.mq.jms.context.WMQInitialContextFactory{*}"/>
+                    <filecontent filename="{*}.xml" pattern="{*}&quot;com.ibm.mq.jms.context.WMQInitialContextFactory&quot;{*}"/>
+                </or>
+                -->
+                <filecontent filename="{*}.{ext}" pattern="{*}com.ibm.mq.jms.context.WMQInitialContextFactory{*}"/>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ JMS InitialContextFactory">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/jms/MQeQueueConnection.html"/>
+                </classification>
+                <hint effort="1">
```

Comment:
Is this really a trivial change? Seems like this might be a "3" - "Complex migration effort with documented solution"